contracts/contracts/AudiusData.sol

```diff
@@ -0,0 +1,124 @@
+pragma solidity ^0.5.0;
+
+
+import "./SigningLogicInitializable.sol";
+
+
+/** @title Contract for Audius User Data management */
+contract AudiusData is SigningLogicInitializable {
+
+    /// @notice Address permissioned to verify users
+    address verifierAddress;
+
+    event ManageEntity(
+        uint _userId,
+        address _signer,
+        string _entityType,
+        uint _entityId,
+        string _metadata,
+        string _action
+    );
+
+    event ManageIsVerified(
+        uint _userId,
+        bool _isVerified
+    );
+
+    bytes32 constant MANAGE_ENTITY_REQUEST_TYPEHASH = keccak256(
+        "ManageEntity(uint userId,string entityType,uint entityId,string action,string metadata,bytes32 nonce)"
+    );
+
+    function initialize(
+        address _verifierAddress,
+        uint _networkId
+    ) public initializer
+    {
+        require(
+            _verifierAddress != address(0x00),
+            "Must provide verifier address"
+        );
+        verifierAddress = _verifierAddress;
+        SigningLogicInitializable.initialize(
+            "Audius Data",
+            "1",
+            _networkId
+        );
+    }
+
+    /// @notice Manage an entity (ex. User/Track/Playlist/Follow/Repost etc.)
+    /// @param _userId User performing action
+    /// @param _entityType Entity type being manipulated, ex. User/Track/Playlist
+    /// @param _entityId Entity id being manipulated. For new entities, ID assignment will be handled by submitter
+    /// @param _action Action being performed, ex. Create/Update/Delete
+    /// @param _metadata Metadata associated with action
+    /// @param _nonce Nonce used client side for signature generation
+    /// @param _subjectSig Signed data used to recover operation signer
+    function manageEntity(
+        uint _userId,
+        string calldata _entityType,
+        uint _entityId,
```

Comment:
per @jowlee feedback - consider making this a string type for flexibility or potentially adding a separate optional "entityLabel"

Reply:
going to leverage the existing "metadata" field here as adding another string type bumps the call stack limit and offers the same functionality. details will be in code